src/nlp/api.jl

```diff
@@ -116,10 +116,10 @@ function jth_sparse_congrad end
 
 Evaluate ``f(x)`` and ``c(x)`` at `x`.
 """
-function objcons(nlp, x)
+function objcons(nlp::AbstractNLPModel{T, S}, x::AbstractVector) where {T, S}
   @lencheck nlp.meta.nvar x
   f = obj(nlp, x)
-  c = nlp.meta.ncon > 0 ? cons(nlp, x) : eltype(x)[]
+  c = nlp.meta.ncon > 0 ? cons(nlp, x) : S(undef, 0)
```

Comment:
```suggestion
  c = cons(nlp, x)
```
`cons()` already does the right thing here.